fre/make/create_makefile_script.py

```diff
@@ -1,31 +1,41 @@
 '''
-TODO doc-string
+Create the Makefile
 '''
 
 import os
 import logging
-fre_logger = logging.getLogger(__name__)
-
-from pathlib import Path
 
-from .gfdlfremake import makefilefre, varsfre, targetfre, yamlfre
 import fre.yamltools.combine_yamls_script as cy
+from .gfdlfremake import makefilefre, varsfre, targetfre, yamlfre
+
+fre_logger = logging.getLogger(__name__)
 
 def makefile_create(yamlfile, platform, target):
-    """
-    Create the makefile
+    """Create the makefile
+    
+    :param yamlfile: Model compile YAML file
+    :type yamlfile: str
+    :param platform: FRE platform
+    :type platform: str
+    :param target: Predefined FRE targets
+    :type target: str
+    :raises ValueError: Error if platform passed does not exist in platforms yaml configuration 
+
+    .. note:: If additional libraries are defined in the compile.yaml file:
+                - for a container build, a linkline script will be generated
+                  to locate those libraries inside the container and populate 
+                  the Makefile with the correct linker flags
+                - for a bare-metal build, linker flags defined in the yaml
+                  configuration are added to the link line in the Makefile
```

Comment:
maybe `include` and `library` paths are better terms than linker flag.
I associate linker flags as flags passed to the linker.  But maybe include and library paths are linker flags..